std_execution.bs

```diff
@@ -553,6 +553,446 @@ merged into a single one in an asynchronous fashion.
+
+For example, we can compose the algorithms mentioned above so that child operations
+are cancelled when any one of the multiple cancellation conditions occurs:
+<pre highlight=c++>
+sender auto composed_cancellation_example(auto query) {
+  return stop_when(
+    timeout(
+      when_all(
+        first_successful(
+          query_server_a(query),
+          query_server_b(query)),
+        load_file("some_file.jpg")),
+      5s),
+    cancelButton.on_click());
+}
+</pre>
+
+In this example, if we take the operation returned by `query_server_b(query)`, this operation will
+receive a stop-request when any of the following happens:
+* `first_successful` algorithm will send a stop-request if `query_server_a(query)` completes successfully
+* `when_all` algorithm will send a stop-request if the `load_file("some_file.jpg")` operation completes with an error or done result.
+* `timeout` algorithm will send a stop-request if the operation does not complete within 5 seconds.
+* `stop_when` algorithm will send a stop-request if the user clicks on the "Cancel" button in the user-interface.
+* The parent operation consuming the `composed_cancellation_example()` sends a stop-request
+
+
+Note that within this code there is no explicit mention of cancellation, stop-tokens, callbacks, etc.
+yet the example fully supports and responds to the various cancellation sources.
+
+The intent of the design is that the common usage of cancellation in sender/receiver-based code is
+primarily through use of concurrency algorithms that manage the detailed plumbing of cancellation
+for you. Much like algorithms that compose senders relieve the user from having to write their own
+receiver types, algorithms that introduce concurrency and provide higher-level cancellation semantics
+relieve the user from having to deal with low-level details of cancellation.
+
+### Cancellation design summary ###
+
+The design of cancellation described in this paper is built on top of and extends the `std::stop_token`-based
+cancellation facilities added in C++20.
+
+At a high-level, the facilities proposed by this paper for supporting cancellation include:
+* Add `std::stoppable_token` and `std::stoppable_token_for` concepts that generalise the interface of `std::stop_token` type to allow other types with different implementation strategies.
+* Add `std::unstoppable_token` concept for detecting whether a `stoppable_token` can never receive a stop-request.
+* Add `std::in_place_stop_token`, `std::in_place_stop_source` and `std::in_place_stop_callback<CB>` types that provide a more efficient implementation of a stop-token for use in structured concurrency situations.
+* Add `std::never_stop_token` for use in places where you never want to issue a stop-request
+* Add `std::execution::get_stop_token()` CPO for querying the stop-token to use for an operation from its receiver.
+* Add `std::execution::stop_token_type_t<T>` for querying the type of a stop-token returned from `get_stop_token()`
```

Comment:
This being just `decltype(get_stop_token())` right?

Reply:
It's `std::remove_cvref_t<decltype(get_stop_token())>`